snuba/subscriptions/verifier.py

```diff
@@ -51,14 +51,19 @@ def is_off_by_one(self, other: SubscriptionResultData) -> bool:
         data_a = a.pop("data")
         data_b = b.pop("data")
 
-        if (
-            a == b
-            and isinstance(data_a, int)
-            and isinstance(data_b, int)
-            and abs(data_a - data_b) == 1
-        ):
-            return True
-
+        # Data is a list of dicts
+        try:
+            if len(data_a) == len(data_b):
+                for (a, b) in zip(data_a, data_b):
+                    a_keys = a.keys()
+                    b_keys = b.keys()
+                    if a_keys == b_keys:
+                        for key in a_keys:
+                            if abs(a[key] - b[key]) > 1:
```

Comment:
Just to be clear, if a single result is off by more than one, the entire result will be marked as False. Is that intended?

Reply:
Yup. True -> Off by only one (or less). False -> Off by more than one.